runtime/compiler/runtime/MetaData.cpp

```diff
@@ -1077,7 +1081,11 @@ populateBodyInfo(
    //
    if (recompInfo)
       {
-      if (vm->isAOT_DEPRECATED_DO_NOT_USE())
+      if (vm->isAOT_DEPRECATED_DO_NOT_USE()
+#if defined(JITSERVER_SUPPORT)
+         || comp->getPersistentInfo()->getRemoteCompilationMode() == JITServer::SERVER
```

Comment:
Since we have the comp object, a better test could be comp->isOutProcessCompilation()